powerline/bindings/i3/powerline-bar.py

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python
+# vim:fileencoding=utf-8:noet
+from __future__ import (unicode_literals, division, absolute_import, print_function)
+
+import sys
+import time
+import i3
+from threading import Lock
+from powerline import Powerline
+from powerline.lib.monotonic import monotonic
+
+if __name__ == '__main__':
+	name = 'wm'
+	if len(sys.argv) > 1:
+		name = sys.argv[1]
+
+	powerline = Powerline( name, renderer_module='bar' )
+	powerline.update_renderer()
+
+	interval = 0.5
+	lock = Lock()
+
+	def encode( str ): return str.encode('utf-8')
+
+	if sys.version_info > (3,0):
+		def encode( str ): return str
+
+	def render(event=None, data=None, sub=None):
+		global lock
+		with lock:
+			ln = '%{l}'
+			ln += powerline.render(side='left')
+			ln += '%{r}'
+			ln += powerline.render(side='right')
+			print( encode(ln) )
```

Comment:
For some reason it hex-encoded every byte, whether it was ASCII or not and somehow didn't work in general. Output was like this for example:

    \xc2\xa0\xe2\x8c\x9a\xc2\xa020:43%{F-B--u}%{F#ffd0d0d0}%{B#ff303030}\xc2\xa0\xee\x82\xb2

We tried a few things (I dont have py3 installed, friend of me had) and that was the only solution that worked for both py2 and 3